### PR TITLE
feat(prometheus): extra_labels of prometheus plugin_attr support const

### LINKED ISSUE #8146: help request: what extra_labels of prometheus plugin i can use?
### Description

Hi, i want to add some labels to metrics, and some labels are literals instead of vars. I find it will be an empty string when i use literals.  
Besides, i cannot find how to define vars like $upstream_addr. 
```yaml
plugin_attr:
  prometheus:
    metrics:
        http_status:
            extra_labels:
                - upstream_addr: $upstream_addr
                - upstream_status: $upstream_status

```
Any reply would be greatly appreciated :)

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by xinxinh2020**
### Description

Fixes #8146 (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by tzssangglass** [comment]
Why support const? What scenarios require monitoring a constant value?

**Comment by xinxinh2020** [comment]
> Why support const? What scenarios require monitoring a constant value?

In my case, i need to identify which apisix cluster the metrics come from。and every apisix clusters，which are managed by  k8s operator,  have it own labels(such as which k8s CR it belongs to). I think it would be useful to tag these labels in metrics to make us more easy to query metrics of specific custom labels.

**Comment by tzssangglass** [comment]
I don't think it's a good idea to support constants on `extra_labels`, we have metrics like `apisix_node_info`, maybe we should extend it to support more constants.

**Comment by xinxinh2020** [comment]
> I don't think it's a good idea to support constants on `extra_labels`, we have metrics like `apisix_node_info`, maybe we should extend it to support more constants.

As far as i am concerned, `apisix_node_info` is used to show all of the apisix info,  and the `apisix_node_info` is not associated with other metrics like `http_status`. 
Maybe we can add a variable like $node_info_id, so we can refer the `apisix_node_info` in other metrics. 
But i still argue that adding apisix-cluster-level const in some metrics is useful. For example, we have four apisix clusters with lables:
- cluster1: deploy=vm,  provider=aws, type=public
- cluster2: deploy=k8s, provider=aws, type=private
- cluster3: deploy=k8s, provider=azure, type=private
- cluster4: deploy=vm, provider=azure, type=public

Even with extended `apisix_node_info`, how can i query metrics group by  any combination of these labels, such as provider=aws, or deploy=vm&type=public ?

If we tag these labels in metrics, it would be quite convenient and a promql like `apisix_http_status{provider="aws"}` would be enough.

**Comment by tzssangglass** [comment]
> Even with extended `apisix_node_info`, how can i query metrics group by any combination of these labels, such as provider=aws, or deploy=vm&type=public ?

Understood, then you need to add test cases and update the documentation.

**Comment by xinxinh2020** [comment]
> 
That's great. 
And how to add a test case in apisix? Any example ?

**Comment by xinxinh2020** [comment]
> > Even with extended `apisix_node_info`, how can i query metrics group by any combination of these labels, such as provider=aws, or deploy=vm&type=public ?
> 
> Understood, then you need to add test cases and update the documentation.

That's great.
And how to add a test case in apisix? Any example ?

**Comment by tzssangglass** [comment]
> That's great.
> And how to add a test case in apisix? Any example ?

ref: https://github.com/apache/apisix/blob/master/docs/en/latest/plugin-develop.md#write-test-case

**Comment by xinxinh2020** [comment]
>

**Comment by xinxinh2020** [comment]
> > That's great.
> > And how to add a test case in apisix? Any example ?
> 
> ref: https://github.com/apache/apisix/blob/master/docs/en/latest/plugin-develop.md#write-test-case

Test case & doc are provided. Pls review it :)

File: apisix/plugins/prometheus/exporter.lua [hunk 1]
```diff
@@ -80,9 +80,14 @@ local function extra_labels(name, ctx)
         for _, kv in ipairs(labels) do
             local val, v = next(kv)
             if ctx then
-                val = ctx.var[v:sub(2)]
-                if val == nil then
-                    val = ""
+                if v:find("$",1, true) then
```
Comment (spacewander) [inline apisix/plugins/prometheus/exporter.lua:83]: Will "a$b" match?

Comment (xinxinh2020) [reply apisix/plugins/prometheus/exporter.lua:83]: not, "a$b" will be considered as a const now

Comment (spacewander) [reply apisix/plugins/prometheus/exporter.lua:83]: @xinxinh2020 
This branch will match `a$b`.

Comment (spacewander) [reply apisix/plugins/prometheus/exporter.lua:83]:
``` 
￥ cat /tmp/x.lua
#!/usr/bin/env luajit

local v = "a$b"
print(v:find("$",1, true))

 ￥ /tmp/x.lua
2       2
```

Comment (xinxinh2020) [reply apisix/plugins/prometheus/exporter.lua]: > @xinxinh2020 This branch will match `a$b`.

Thinks for reminding.  I fix it already.

File: apisix/plugins/prometheus/exporter.lua [hunk 2]
```diff
@@ -80,9 +80,14 @@ local function extra_labels(name, ctx)
         for _, kv in ipairs(labels) do
             local val, v = next(kv)
             if ctx then
-                val = ctx.var[v:sub(2)]
-                if val == nil then
-                    val = ""
+                if v:find("$",1, true) == 1 then
```
Comment (spacewander) [inline apisix/plugins/prometheus/exporter.lua:83]: Would be better to use string.byte which is O(1).

Comment (xinxinh2020) [reply apisix/plugins/prometheus/exporter.lua]: good advice & work done

**Pull request closed without merging**